src/server/modules/threat/FileCheckController.ts

```diff
@@ -35,13 +35,54 @@ export class FileCheckController {
       )
       return
     }
-
     if (file) {
       if (!(await this.fileTypeFilterService.hasAllowedType(file))) {
         res.unsupportedMediaType(jsonMessage('File type disallowed.'))
         return
       }
+    }
+
+    next()
+  }
+
+  public csvExtensionCheck: (
```

Comment:
The method seems to have quite abit of reused code. Instead of creating a special function to validate csv, what about [passing a parameter to the middleware](https://tsmx.net/express-middleware-function-with-custom-parameters/) e.g. `FileCheckController.fileExtensionCheck('csv')`